db_migrations/0015_recreate_media_files.sql

```diff
@@ -0,0 +1,44 @@
+-- Migration 0015: Recreate media_files table with clean schema for media cache
+--
+-- This migration drops the experimental media_files table from migration 0002
+-- and recreates it with a proper schema designed for:
+-- 1. Group profile images (MIP-01)
+-- 2. Future group media files (MIP-04)
+--
+-- Key features:
+-- - Cross-account cache with reference tracking
+-- - Normalized columns (no JSONB)
+-- - Hash-based storage organization
+-- - LRU eviction support via accessed_at
+
+-- Drop old experimental media_files table
+DROP TABLE IF EXISTS media_files;
+
+-- Create new media_files table with clean schema
+CREATE TABLE media_files (
+    id INTEGER PRIMARY KEY AUTOINCREMENT,
+    mls_group_id BLOB NOT NULL,
+    account_pubkey TEXT NOT NULL,
+    file_path TEXT NOT NULL,
+    file_hash TEXT NOT NULL,          -- Hex-encoded hash of ENCRYPTED data (image_hash for group images, per MIP-01)
+    mime_type TEXT NOT NULL,          -- Canonical MIME type (e.g., "image/jpeg")
+    media_type TEXT NOT NULL,         -- "group_image" or "chat_media"
+    blossom_url TEXT,                 -- Optional: NULL for default server, set for custom
+    nostr_key TEXT,                   -- Optional: For future MIP-04 cleanup
+    dimensions TEXT,                  -- Optional: Image/video dimensions as "widthxheight" (e.g., "1920x1080")
+    blurhash TEXT,                    -- Optional: BlurHash string for progressive loading
+    created_at INTEGER NOT NULL,
+    accessed_at INTEGER NOT NULL,     -- For LRU eviction
+    FOREIGN KEY (account_pubkey) REFERENCES accounts(pubkey) ON DELETE CASCADE
+);
```

Comment:
Previously we had a `file_metadata JSONB` field. One of it purposes (if I recall correctly) was to store the original filename. I think we should still have that.

Things like dimensions and blurhash I believe should be in there as well. The reason being that those are metadata 'attributes' that only make sense on certain types of files, namely images, so I think it kinda maps better to an unstructured json field.